aiida/cmdline/commands/cmd_profile.py

```diff
@@ -52,9 +52,13 @@ def profile_list():
 
 
 @verdi_profile.command('show')
-@arguments.PROFILE(required=False, callback=defaults.get_default_profile)
-def profile_show(profile):
+@arguments.PROFILE(required=False)
+@click.pass_context
+def profile_show(ctx, profile):
     """Show details for PROFILE or, when not specified, the default profile."""
+    if profile is None:
```

Comment:
well in `setdefault` I make it required, because without value the command does not make sense. So there I am guaranteed that `profile is not None`. However, `verdi profile show` does not require a profile, so the `profile` will be `None` if not specified. However, in that case `ctx.obj.profile` _should_ have been defined in `verdi` itself. But just to be sure I keep the check in case no profile exists yet, for example.

Reply:
> However, in that case ctx.obj.profile should have been defined in verdi itself. But just to be sure I keep the check in case no profile exists yet, for example.

Not quite sure I follow the logic here - if no profile exists, wouldn't `ctx.obj.profile` be `None` as well?

Reply:
Yes, exactly, since `ctx.obj.profile` is not guaranteed to contain a profile, for example in the case when no profiles exist yet, I don't want `verdi process show` to except, so therefore I have to check for `None`